hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -325,56 +310,16 @@ private HoodieData<HoodieRecord<T>> readRecordsForGroup(JavaSparkContext jsc, Ho
   private HoodieData<HoodieRecord<T>> readRecordsForGroupWithLogs(JavaSparkContext jsc,
                                                                   List<ClusteringOperation> clusteringOps,
                                                                   String instantTime) {
-    HoodieWriteConfig config = getWriteConfig();
-    HoodieTable table = getHoodieTable();
-    // NOTE: It's crucial to make sure that we don't capture whole "this" object into the
-    //       closure, as this might lead to issues attempting to serialize its nested fields
-    StorageConfiguration<?> storageConf = table.getStorageConf();
-    HoodieTableConfig tableConfig = table.getMetaClient().getTableConfig();
-    String bootstrapBasePath = tableConfig.getBootstrapBasePath().orElse(null);
-    Option<String[]> partitionFields = tableConfig.getPartitionFields();
-
     int readParallelism = Math.min(writeConfig.getClusteringGroupReadParallelism(), clusteringOps.size());
 
     return HoodieJavaRDD.of(jsc.parallelize(clusteringOps, readParallelism).mapPartitions(clusteringOpsPartition -> {
       List<Supplier<ClosableIterator<HoodieRecord<T>>>> suppliers = new ArrayList<>();
       clusteringOpsPartition.forEachRemaining(clusteringOp -> {
 
         Supplier<ClosableIterator<HoodieRecord<T>>> iteratorSupplier = () -> {
-          long maxMemoryPerCompaction = IOUtils.getMaxMemoryPerCompaction(new SparkTaskContextSupplier(), config);
```

Comment:
These deleted code simply moved to the SparkJobExecutionStrategy used to provide a common reading method